peps/pep-9999.rst

```diff
@@ -0,0 +1,193 @@
+no consensus has been reached about many contentious issues,
+including, but not limted to:
+
+- Conventions for designing new API functions;
+- how to deal with compatibility;
+- what's the best strategy for handling errors;
+- the future of the Stable ABI and the Limited API;
+- whether to switch to a handle-based API convention (and how).
+
+The general feeling is that there are too many stakeholders,
+proposals, requirements, constraints, and conventions,
+to make progress without having a small trusted group of deciders.
+
+At the 2023 Language Summit in Salt Lake City it was decided to start work on
+an `inventory of problems <https://github.com/capi-working-group/problems>`__.
+At the 2023 core developer sprint in Brno this work is more or less finished
+and after a discussion it appeared that the next step is to establish
+a working group to ensure that we're not stymied forever.
+
+The Steering Council has
+`indicated <https://github.com/python/steering-council/issues/201#issuecomment-1648848155>`__
+its desire to delegate decisions about the C API
+to such a working group, anticipating its formal establishment.
+
+Specification
+=============
+
+We propose the creation of a new group, the C API Working Group.
+This group will be responsible for overseeing and coordinating the development and
+maintenance of the Python C API.
+It will do this by establishing the principles underpinning the work
+and publishing guidelines that the core developers can refer to.
+
+The "operations and process" section below describes
+how the working group operates and how it is governed.
+
+Mandate
+-------
+
+The working group's mandate is to ensure that the Python C API is:
+
+* **Useful**: It should provide access to all relevant functionality.
+* **Ergonomic**: It should be easy to use.
+* **Fair**: It should serve the purposes of a variety of stakeholders.
+* **Stable**: It should avoid requiring users to update their code more than necessary.
+* **Evolvable**: It should give Python implementers enough freedom to change implementation details.
+* **Maintainable**: It should be easy to maintain and extend.
+
+For several of these items, a more comprehensive listing of constraints
+can be found in the PEP (number TO DO) which is based on the 
+`inventory of C API problems <https://github.com/capi-working-group/problems>`__.
+
+The stability requirement is not meant to prevent sweeping C API changes ---
+one of the working group's initial tasks is to settle the question of
+whether we intend any landmark changes to the C API in the coming five years,
+and if so, to select or write a suitable proposal.
+However, the working group is required to carefully consider the migration
+efforts required of package maintainers and other stakeholders.
+
+Operations and process
+----------------------
+
+The working group has at least three members,
+comprised of prominent Python core developers.
```

Comment:
Maybe "active" rather than prominent? Or some word to imply that we are able to commit time and effort towards it?

Reply:
I took "prominent" (but not the rest of the sentence) from Jelle's PEP 729. And being active doesn't always translate into suitability.

There's no way to enforce time and effort taken -- we all do our best, and if a member fails to show up repeatedly, the rest of us have to make a point of replacing that member. But I don't think that needs to be written down. The in-person discussion about the Editorial Board today indicated that the SC (or at least Thomas) thinks committees are best positioned to decide on their own renewal policies.

IOW I think "prominent" is right here. The real sting in this sentence is in "core developers". Jelle's PEP has "community members". But I feel that since almost all C API work ends up on the shoulders of the core devs, it's fair that the WG is comprised of core devs. (This is different from typing, where a lot of the work is a burden for 3rd party tooling maintainers.) I do have some words about listening to stakeholders.

Reply:
After spending a bit more time thinking about it, I think my concern with "prominent" is that it doesn't really give someone a self-development path that would lead here. That said, I encourage virtually everyone I mentor to "become more prominent", because it is the reality of how these things work out, but I'm just a little uncomfortable with making it a formal requirement.

I'm far less concerned about the core developers part, because that's already a reasonably well-defined aspiration (even if, in practice, being prominent is a sizable part of it).

But I feel I've already dedicated too much thinking on this point, and am forcing you to spend too much time reading my thoughts :-) So I'll leave it